asreview/webapp/api/projects.py

```diff
@@ -1166,6 +1166,13 @@ def api_export_dataset(project):
         # read the dataset into a ASReview data object
         as_data = read_data(project)
 
+        # Retrieve prior information from state
+        with open_state(project.project_path) as s:
+            priors = s.get_priors()["record_id"].tolist()
+
+        # Add a new column 'is_prior' to the dataset
+        as_data.df['is_prior'] = as_data.df.index.isin(priors).astype(int)
```

Comment:
isin can introduce performance issues. It is better to avoid it. We can make use of a `join` similar to the notes below. 

You might be able to avoid the open_state call (expensive) if you reuse state_df.

Reply:
Yeah, instead of opening the state again you could do (didn't test this):
```[python]
state_df["is_prior"] = state_df.query_strategy.eq("prior").astype(int)
as_data.df = as_data.df.join(state_df["is_prior"], on="record_id")
```

Reply:
![image](https://github.com/asreview/asreview/assets/36502709/2f1ff1ef-efc4-4df8-bd08-0149561b6453)

Reply:
@PeterLombaers  your suggestion seems do work!